raiden/tests/integration/contracts/fixtures/contracts.py

```diff
@@ -13,42 +13,54 @@
 )
 
 
-@pytest.fixture
-def deploy_contract(deploy_client, contract_manager):
-    """Deploy a contract using raiden-contracts contract manager"""
-    def f(contract_name: str, args=None):
-        if args is None:
-            args = []
-        compiled = {
-            contract_name: contract_manager.get_contract(contract_name),
-        }
-        return deploy_client.deploy_solidity_contract(
-            contract_name,
-            compiled,
-            constructor_parameters=args,
-        )
-    return f
-
-
-@pytest.fixture
-def secret_registry_contract(deploy_contract):
-    return deploy_contract(CONTRACT_SECRET_REGISTRY)
+@pytest.fixture(name='secret_registry_contract')
+def deploy_secret_register_and_return_jsonrpc_proxy(deploy_client, contract_manager):
```

Comment:
- Why set the name over the attribute?
- s/deploy_secret_register/deploy_secret_registry

Reply:
> Why set the name over the attribute?

It allows for more descriptive function names in fixture implementation, while allowing to _use_ the fixture with a telling argument name.

Reply:
@palango it was a suggestion from our test review, please refer to the issue: https://github.com/raiden-network/raiden/issues/3346

Reply:
It's quite nice, and I didn't know that was possible.